### PR TITLE
Update config and pipeline of DIC

### PR DESCRIPTION
**Pull request by Yshuo-Li**
## Motivation
The prior pipeline of DIC loads facial landmarks from files while generating from images by face_alignment will be better.

## Modification
- Add GenerateLandmark
- Update config and pipeline of DIC

### CODE REVIEW

File: configs/restorers/dic/dic_gan_x8c48b6_g4_500k_CelebAHQ.py [hunk 1]
```diff
@@ -111,18 +112,17 @@
...
             type=train_dataset_type,
-            gt_folder='data/celeba-hq/train/',
-            ann_file='data/celeba-hq/train_info_list_256.npy',
+            gt_folder='data/CelebA-HQ/train_256/all_256',
             pipeline=train_pipeline,
             scale=scale)),
     val=dict(
         type=val_dataset_type,
-        gt_folder='data/celeba-hq/val/',
+        gt_folder='data/CelebA-HQ/valid_256',
```
Comment (innerlee) [inline configs/restorers/dic/dic_gan_x8c48b6_g4_500k_CelebAHQ.py:120]: Any doc on the change of path?

**Comment by innerlee** [comment]
@ckkelvinchan please review the `GenerateLandmark` part

File: tools/data/inpainting/celeba-hq/README.md [hunk 2]
```diff
@@ -19,7 +19,7 @@ mmediting
 ├── tools
 ├── configs
 ├── data
-│   ├── celeba-hq
+│   ├── CelebA-HQ
```
Comment (innerlee) [inline tools/data/inpainting/celeba-hq/README.md:22]: We prefer lower-cased file names, so please change it back

File: mmedit/datasets/pipelines/generate_assistant.py [hunk 3]
```diff
@@ -167,3 +167,75 @@ def __repr__(self):
...
+
+        Returns:
+            landmark (Tuple[float]): Location of landmark.
+        """
+
+        try:
+            import face_alignment
+        except ImportError:
+            raise ImportError('please import face-alignment.')
+        if not self.init:
```
Comment (innerlee) [inline mmedit/datasets/pipelines/generate_assistant.py:219]: `self.face_alignment_model is None` is good enough to check initialization

File: mmedit/datasets/pipelines/generate_assistant.py [hunk 4]
```diff
@@ -167,3 +167,75 @@ def __repr__(self):
         repr_str += (f'sample_quantity={self.sample_quantity}, '
                      f'scale={self.scale}, target_size={self.target_size}')
         return repr_str
+
+
+@PIPELINES.register_module()
+class GenerateLandmark:
```
Comment (ckkelvinchan) [inline mmedit/datasets/pipelines/generate_assistant.py:173]: How about `GenerateFacialLandmarks`?

File: mmedit/datasets/pipelines/generate_assistant.py [hunk 5]
```diff
@@ -167,3 +167,75 @@ def __repr__(self):
...
+
+        Args:
+            image (numpy.ndarray): Face image.
+
+        Returns:
+            landmark (Tuple[float]): Location of landmark.
+        """
+
+        try:
+            import face_alignment
```
Comment (ckkelvinchan) [inline mmedit/datasets/pipelines/generate_assistant.py:216]: This line would be called whenever `_face_alignment_detector` is called. 

Suggest to import only once as in [here](https://github.com/open-mmlab/mmediting/blob/master/mmedit/apis/restoration_face_inference.py#L8).

File: mmedit/datasets/pipelines/generate_assistant.py [hunk 6]
```diff
@@ -167,3 +167,75 @@ def __repr__(self):
...
+            self.face_alignment_model = face_alignment.FaceAlignment(
+                face_alignment.LandmarksType._2D,
+                device=self.device,
+                flip_input=False)
+            self.init = True
+        faces = self.face_alignment_model.get_landmarks(image)
+        index = 0
+        max_size = 0
+        for i, face in enumerate(faces):
+            size = face[8, 1] - face[19, 1]
```
Comment (ckkelvinchan) [inline mmedit/datasets/pipelines/generate_assistant.py:229]: Suggestion:  Provide explanations on the indices.

File: mmedit/datasets/pipelines/generate_assistant.py [hunk 7]
```diff
@@ -167,3 +167,75 @@ def __repr__(self):
...
+            if size > max_size:
+                max_size = size
+                index = i
+        landmark = faces[index]
```
Comment (ckkelvinchan) [inline mmedit/datasets/pipelines/generate_assistant.py:233]: It seems that this function returns the largest face if multiple faces exist. Better to mention that in the docstrings.

**Pull request closed without merging**